cwl/src/main/scala/cwl/package.scala

```diff
@@ -60,6 +61,8 @@ package object cwl extends TypeAliases {
   type Expression = ECMAScriptExpression :+: ECMAScriptFunction :+: CNil
 
   object StringOrExpression {
+    def unapply(listing: Array[IwdrListingArrayEntry] :+: StringOrExpression :+: CNil): Option[StringOrExpression] = listing.select[StringOrExpression]
```

Comment:
that coproduct feels like it should be a named `type`, which hopefully would make this custom `unapply` less surprising

Reply:
Good idea, it also simplifies the `listing` field of the `InitialWorkDirRequirement` case class